### PR TITLE
Simple fix to issue #80

### PR DESCRIPTION
**Pull request by favre49**
This is a simple fix to issue #80 . This is done by adding small values to the diagonal of the covariance matrix whenever the Cholesky decomposition fails to make the matrix positive definite. According to my testing, this seems to have fixed the problem.

### CODE REVIEW

File: doc/optimizers.md [hunk 1]
```diff
@@ -506,6 +506,47 @@ optimizer.Optimize(f, coordinates);
...
+DE optimizer(200, 1000, 0.6, 0.8);
+optimizer.Optimize(f, coordinates);
+```
+
+#### See also:
+
+ * [Differential Evolution - A simple and efficient adaptive scheme for global optimization over continuous spaces](http://www1.icsi.berkeley.edu/~storn/TR-95-012.pdf)
+ * [Differential Evolution in Wikipedia](https://en.wikipedia.org/wiki/Differential_Evolution)
+ * [Arbitrary functions](#arbitrary-functions)
+
```
Comment (rcurtin) [inline doc/optimizers.md]: Oops, I think these changes should not be a part of this PR. :)

File: include/ensmallen_bits/cmaes/cmaes_impl.hpp [hunk 2]
```diff
@@ -128,7 +128,20 @@ double CMAES<SelectionPolicyType>::Optimize(
...
+      covLower = arma::chol(C.slice(idx0), "lower");
+    }
+    catch (const std::runtime_error& err)
+    {
+      // If the covariance matrix isn't positive definite, add a small amount
+      // to the diagonal and try again
+      arma::mat I(iterate.n_elem, iterate.n_elem, arma::fill::eye);
+      C.slice(idx0) += I*(1e-16);
+      covLower = arma::chol(C.slice(idx0), "lower");
+    }
```
Comment (rcurtin) [inline include/ensmallen_bits/cmaes/cmaes_impl.hpp]: Actually you can simplify this a bit...

```
C.slice(idx0).diag() += 1e-16;
```

Also, if `1e-16` is generally sufficient I'm okay with that, but it's also possible to loop iteratively until the Cholesky decomposition succeeds.  You could use the version `chol(C.slice(idx0), covLower, "lower")` which returns a `bool`, and loop until it's true.

File: include/ensmallen_bits/cmaes/cmaes_impl.hpp [hunk 3]
```diff
@@ -281,4 +294,4 @@ double CMAES<SelectionPolicyType>::Optimize(
 
 } // namespace ens
 
-#endif
+#endif
```
Comment (rcurtin) [inline include/ensmallen_bits/cmaes/cmaes_impl.hpp]: No need to remove the newline from the end of the file, please. :+1:

File: include/ensmallen_bits/cmaes/cmaes_impl.hpp (hunk 2 above)
Comment (rcurtin) [reply include/ensmallen_bits/cmaes/cmaes_impl.hpp]: Here is a simplification. :)

```
while (!arma::chol(covLower, C.slice(idx0), "lower"))
  C.slice(idx0).diag() += 1e-16;
```

Comment (favre49) [reply include/ensmallen_bits/cmaes/cmaes_impl.hpp]: Yup, far simpler, thank you for the suggestion

**Comment by rcurtin** [comment]
@favre49 thanks! :+1:

**Pull request merged**